app/(server)/_features/user/schema.ts

```diff
@@ -0,0 +1,21 @@
+import { serial, text, integer, pgTable, timestamp } from 'drizzle-orm/pg-core';
+import { type InferInsertModel, type InferSelectModel, sql } from 'drizzle-orm';
+
+export const users = pgTable('users', {
+  id: serial('id').primaryKey(),
+  email: text('email').notNull().unique(),
+  name: text('name').notNull(),
+  pictureUrl: text('picture_url'),
+  whitelistFeature: text('whitelist_feature')
+    .array()
+    .notNull()
+    .default(sql`array[]::text[]`),
+  accountId: integer('account_id'),
```

Comment:
this should be unique & not null ?

Reply:
@GagahGK 
I also thought this `accountId` should be unique and not null. However, this account ID is related only to the inlive API data. What if in the future we decide inlive room will use its own authentication and no longer depends to the inlive API data? Do we still need this `accountId` field? I think the answer is no.

Besides that, I still don't know the usage of this `accountId` field except for tracking the account ID from inlive API. You will see I only stored the `accountId` and not using it anywhere. Also, the email is already unique enough and already connected with the `accountId` inlive API data. So, I actually don't think this `accountId` has any purpose except for tracking purposes and can be replaced by email easily.

So, my conclusion is, it is fine to store the `accountId` field. However, this is not required field because there will be a chance in the future this field is no longer needed.

Reply:
while it's true that accountID in the future may no longer needed in the future, it doesn't mean that a user can have a duplicates accountID with other user

since currently we authenticate user by token that was given by inliveAPI, with this in mind, this means that an inlive room user is tied to an account from inliveAPI

to identify a user we authenticate the token and then search the user by the account id from the token / inliveAPI endpoint

we need to enforce those rules in the database, what if there was a bad query, migration that updates the table resulting in a multiple users having the same accountID ? we might granting access to a wrong user, letting them have unintended access and possibly do malicious thing 

inlive room might have its own implementation for authentication in the future, if that's the case we can just remove the accountID from the table and update the code to identify user from inlive-api token into inlive-room own token, that's it

Reply:
@GagahGK  I would like to say two things:
1. If it's for the sake of enforcing those rules (account ID is not null and must be a unique) then I think it's fine to follow those rules.
2.  I have a different implementation than this statement.
> to identify a user we authenticate the token and then search the user by the account id from the token / inliveAPI endpoint

To identify a user, we can search the user by the **email** not the account ID. As I said, email is already unique and it's better to search the user by the email as the user table cannot have a different user with duplicate email.
You can [see this line](https://github.com/inlivedev/inlive-room/pull/128/files#diff-0159f45d21818b3aa9048781c97594e4a895c0bf22b1dc713421029425be48b0R67) where I get the current user data by email, [and this line](https://github.com/inlivedev/inlive-room/pull/128/files#diff-656b087b3ad8a5d8363218784086628b4fe73917e14539df30e4b62d9ff6eb76R28) where I check whether an email has existed before actually adding a new user to the user table.

Reply:
> If it's for the sake of enforcing those rules (account ID is not null and must be a unique) then I think it's fine to follow those rules.

yes since the authentication is using token from inliveAPI we need to enforce those rule, because logicallyif the token is valid, then we know that there's a user that registered and that user should be able to be identified by the email and the userID from inliveAPI, querying by email or accountID results in same user

the identified user from that token shouldn't result in different user, it doesn't makes sense

> I have a different implementation than this statement.

There's no problem for using this approach, it works, and also perform what it needed, the usual case is that we will create two different function getUserByEmail() and getUserByAccountID(), but using email is prefered since AccountID will be phased out in the future

---

I hope you understand why we should make sure that the accountID should be unique and avoid having duplicates